fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -114,43 +114,18 @@ fn expand_function_arguments(
     custom_enums: &HashMap<String, Property>,
     custom_structs: &HashMap<String, Property>,
 ) -> Result<(TokenStream, TokenStream), Error> {
-    let n_inputs = fun.inputs.len();
-    // Check that we have the expected types
-    if n_inputs != 4 {
-        return Err(Error::MissingData(format!(
-            "Expected exactly 4 inputs, found {}",
-            n_inputs
-        )));
-    }
-    let given_types: Vec<String> = fun.inputs[..3]
-        .iter()
-        .map(|x| x.type_field.clone())
-        .collect();
-    let expected = ["u64", "u64", "b256"];
-    if given_types != expected {
-        return Err(Error::InvalidType(format!(
-            "Expected the 3 first types to be {:?}, found {:?}",
-            expected, given_types
-        )));
-    };
     // Ignore the first three arguments so we don't have to provide them when calling
     // the contracts' methods
-    let first_index = 3;
     let mut args = Vec::with_capacity(1);
     let mut call_args = Vec::with_capacity(1);
-    // We use a for loop because we expect exactly 4 arguments (so len(fun.inputs)==4)
-    // but we might extend the number of authorized inputs in the future
-    for (i, param) in fun.inputs[first_index..].iter().enumerate() {
+
+    for (i, param) in fun.inputs.iter().enumerate() {
         // For each [`Property`] in a function input we expand:
         // 1. The name of the argument;
         // 2. The type of the argument;
         // Note that _any_ significant change in the way the JSON ABI is generated
         // could affect this function expansion.
         // TokenStream representing the name of the argument
-        if param.type_field == "()" {
```

Comment:
why is this not needed anymore?

Reply:
It's not related to #13 but to #95 and I'm not sure we can just remove it

Reply:
Sway now supports calls with actually 0 arguments (`fn foo()` instead of `fn foo(input: ()))`. So we can remove this check above. See tests with proper empty arguments down below.

Reply:
ok!